run/run.go

```diff
@@ -80,7 +81,17 @@ func GetMain() *cobra.Command {
 		ext.GetOpenAPIFile = func(args []string) (s string, err error) {
 			return filepath.Join(args[0], kptfile.KptFileName), nil
 		}
-		return kptopenapi.ConfigureOpenAPI(f, cmdutil.K8sSchemaSource, cmdutil.K8sSchemaPath)
+		err := kptopenapi.ConfigureOpenAPI(f, cmdutil.K8sSchemaSource, cmdutil.K8sSchemaPath)
+		if err != nil {
+			return err
+		}
+
+		// add openAPI definitions from Kptfile to configured openAPI
+		if addErr := openapi.AddSchemaFromFile(filepath.Join(args[0], kptfile.KptFileName)); addErr != nil {
```

Comment:
So this is already handled by overriding the path to the Krmfile on line 80 and reading the file as part of the command runners in cmd/config. An example for create-setter:
https://github.com/kubernetes-sigs/kustomize/blob/a7545bdad3f35eefa48cdd3980b1c82e11e02956/cmd/config/internal/commands/cmdcreatesetter.go#L106-L113

I don't think we currently do this for the live commands, so we need to address that.

Reply:
@mortent The purpose here is to add it as part of top level Kpt command so that all underlying commands can use it by default without needing to add it explicitly in each command(in your example link).

Reply:
Yeah, I agree we should try to put this logic in one place for all commands. Can we make changes to the command runners so we don't have load the file twice for the same command?

Reply:
@mortent Not sure I followed your comment. Can you point me to the location where you want me to add this? I feel this is the top level central appropriate place to add this logic. Once this is merged, I can remove all the rest of the places which are deep down (like the one in link you mentioned). So by default, if any of the kpt commands are invoked, the openAPI is loaded from one of the 3 sources + from the Kptfile. The underlying commands can just use it and need not load it (like the ones in kustomize).

Reply:
I think we are on the same page. I wanted to see if we could remove the code in the runners that loads the openAPI, and sounds like you already have a plan for that.

Reply:
Hi guys, it looks like this PR breaks master for commands that don't operate on a package directory as the `args` array is empty which results in an out of bounds error.

Reply:
@ashlineldridge thanks for reporting this. We will address it today.